src/sentry/api/event_search.py

```diff
@@ -906,6 +906,76 @@ def validate_aggregate(field, match):
         )
 
 
+FUNCTION_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<columns>[^\)]*)\)$")
+
+FUNCTIONS = {
+    "percentile": {
+        "name": "percentile",
+        "args": [
+            {"name": "column", "type": "numeric_column"},
+            {
+                "name": "percentile",
+                "type": "number",
+                "fn": lambda v: (v > 0 and v < 1, "not between 0 and 1"),
+            },
+        ],
+        "transform": "quantile(%(percentile).2f)(%(column)s)",
```

Comment:
This general structure looks good, you might want to consider using a `namedtuple` or helper object to give a simpler to build interface and better errors as you build more of these functions out.

Reply:
That was the plan. I figured that the necessary structure would become more apparent as we added more functions.